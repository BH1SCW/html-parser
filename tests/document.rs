use html_parser::{HtmlParser, Result};
use indoc::indoc;
use insta::assert_debug_snapshot;

#[test]
fn it_can_parse_minimal_document() -> Result<()> {
    let markup = "<!DOCTYPE html><html></html>";
    let ast = HtmlParser::parse(markup)?;
    assert_debug_snapshot!(ast);
    Ok(())
}
#[test]
fn it_can_parse_document_with_comments() -> Result<()> {
    let markup = indoc!(
        r#"
        <!-- comment -->
        <!-- comment -->
        <!DOCTYPE html>
        <!-- comment -->
        <!-- comment -->
        <html>
        <!-- comment -->
        </html>
        <!-- comment -->
        <!-- comment -->
    "#
    );
    let ast = HtmlParser::parse(markup)?;
    assert_debug_snapshot!(ast);
    Ok(())
}
#[test]
fn it_error_when_doctype_and_multiple_html() {
    let markup = "<!DOCTYPE html><html></html><html></html>";
    assert!(HtmlParser::parse(markup).is_err());
}
