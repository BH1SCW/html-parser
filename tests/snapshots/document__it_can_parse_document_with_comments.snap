---
source: tests/document.rs
expression: dom
---
{
  "treeType": "document",
  "nodes": [
    {
      "name": "html",
      "variant": "normal"
    }
  ]
}
